integration-tests/test_e2e.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+import conftest
+from constants import MACHINE_ID_FILE
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_insights_client_version_in_inventory(insights_client, external_inventory):
+    """
+    Verify running insights-client creates a new host on Inventory and the host has
+    the insights client and egg version information available
+
+        test_steps:
+            1. Register the system to Insights
+            4. Retrieve the system profile from Inventory
+        expected_results:
+            1. Successful registration to insights server
+            2. The system is successfully retrieved from Inventory
+            3. system data retrieved from Inventory has client and egg version info
+    """
+    insights_client.run("--register")
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    with open(MACHINE_ID_FILE, "r") as f:
+        machine_id = f.read()
+    # fetching host details to get host ID
+    response = external_inventory.get(path=f"hosts?insights_id={machine_id}")
```

Comment:
use `external_inventory.this_system()` here instead

Reply:
this_system() gives scope error

Reply:
it works for me; hard to say when no details or reproducer systems are provided...

Reply:
For reference https://issues.redhat.com/browse/CCT-622, 
Let me know if I missed any details.

Reply:
> For reference https://issues.redhat.com/browse/CCT-622

see my comment there ...

Reply:
it should work fine now, please check it again

Reply:
Thanks!